### PR TITLE
added edit image

### PR DESCRIPTION
**Pull request by Davut97**
i added option to upload a new image and fixed some bugs

### CODE REVIEW

File: src/Facebook.js [hunk 1]
```diff
@@ -1,82 +1,138 @@
...
 };
 
-const EditProfile = (props) =>{
-  const [isEditing , setIsEditing] = React.useState(false);
-  
+const EditProfile = (props) => {
+  const [isEditing, setIsEditing] = React.useState(false);
+  const [image, setImage] = React.useState(null);
+  const [url, setUrl] = React.useState('');
+  const [progress, setProgress] = React.useState(0);
```
Comment (louisrli) [inline src/Facebook.js:51]: name this variable `uploadProgress` (progress is ambiguous)

File: src/Facebook.js [hunk 2]
```diff
@@ -1,82 +1,138 @@
...
   );
 };
 
-const EditProfile = (props) =>{
-  const [isEditing , setIsEditing] = React.useState(false);
-  
+const EditProfile = (props) => {
+  const [isEditing, setIsEditing] = React.useState(false);
+  const [image, setImage] = React.useState(null);
+  const [url, setUrl] = React.useState('');
```
Comment (louisrli) [inline src/Facebook.js:50]: what is this the url of? better variable name

File: src/Facebook.js [hunk 3]
```diff
@@ -1,82 +1,138 @@
...
+  const [progress, setProgress] = React.useState(0);
   const showEditInputs = () => {
     setIsEditing(true);
-  }
+  };
+  const handleChange = (e) => {
```
Comment (louisrli) [inline src/Facebook.js:55]: better variable name, is this handling the change of the inputs or what is it handling the change of?

File: src/Facebook.js [hunk 4]
```diff
@@ -1,82 +1,138 @@
...
+    if (e.target.files[0]) {
+      setImage(e.target.files[0]);
+    }
+  };
+  const handleUpload = () => {
+    const uploadTask = storage.ref(`images/${image.name}`).put(image);
```
Comment (louisrli) [inline src/Facebook.js:61]: hm actually i realized i configured the firebase rules wrong. i don't think this works at the moment with the production database, does it?

```

service firebase.storage {
  match /b/{bucket}/o {
    // Files look like: "user/<UID>/path/to/file.txt"
    match /profilePictures/{userId}/{allPaths=**} {
      allow read, write: if request.auth != null && request.auth.uid == userId;
    }
  }
}```

^ see what the expected bucket and such is

**Comment by Davut97** [comment]
I fixed the variables names, and fixed a bug when user clicking upload without an choosing an image the website was crashing, also fixed a bug where if the user clicked submit without entering any info, the city and profile field will be empty.

**Pull request merged**